### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by janowsiany**
### TODO:
- [x] Usage
    - [x] `Setting/getting locale`
    - [x] `Adding translations by methods`
    - [x] `Getting translations`
    - [x] `Creating a React component`
    - [x] `Formatting numbers`
    - [x] `Incremental loading of translations`
- [x] Translations files
    - [x] `JSON`
    - [x] `YAML`
- [ ] Test coverage
    - [ ] `client`
    - [x] `server`

### CODE REVIEW

**Comment by janowsiany** [comment]
Please pay no regard to first result of test coverage it always underestimates the result.

File: i18n.tests.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    it('should support JSON files', async function () {
+        await i18n.setLocale('es-ES')
+        expect(i18n.__('common.name')).to.equal('json');
+    });
+
+    it('should be able to set locale', function () {
+        expect(i18n.setLocale('de-DE')).to.be.ok;
+        expect(i18n.getLocale()).to.equal('de-DE');
+        expect(i18n.setLocale('pl-PL')).to.be.ok;
+        expect(i18n.getLocale()).to.equal('pl-PL').and.not.equal('de-DE');
```
Comment (radekmie) [inline i18n.tests.js]: Unnecessary check.

File: i18n.tests.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        expect(i18n.__('common.name')).to.equal('yml');
+    });
+
+    it('should support JSON files', async function () {
+        await i18n.setLocale('es-ES')
+        expect(i18n.__('common.name')).to.equal('json');
+    });
+
+    it('should be able to set locale', function () {
+        expect(i18n.setLocale('de-DE')).to.be.ok;
```
Comment (radekmie) [inline i18n.tests.js]: It is asynchronous, right?

File: i18n.tests.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        expect(i18n.getCurrencyCodes('pl-PL')).to.deep.equal(["PLN"]);
+    });
+
+    it('should be able to listen on locale change', async function () {
+        const callback = sinon.spy();
+
+        expect(i18n.onceChangeLocale(callback)).to.equal(undefined);
+
+        const result = await i18n.setLocale('pl-PL');
+        // expect(result).to.be.ok;
```
Comment (radekmie) [inline i18n.tests.js]: ?

File: package.js [hunk 4]
```diff
@@ -10,6 +10,17 @@ var npmDependencies = {
...
+var packages = [
+    'http',
+    'webapp',
+    'tracker',
+    'promise',
+    'ecmascript',
+    'underscore',
+    'universe:utilities@2.3.2',
+    'isobuild:compiler-plugin@1.0.0'
+]
```
Comment (radekmie) [inline package.js]: Any reason, why you've done that?

**Comment by cristo-rabani** [comment]
Hello,
Yes I agree with Radek. Please move tests files to separate directory and we can merge test into master branch.

**Comment by janowsiany** [comment]
@cristo-rabani @radekmie I moved majority of tests related files to tests directory. However i left the main file `i18n.tests.js` still in the root directory as without it there is an error prompted in the console. Its message is: `SyntaxError: Unexpected token (`, however i doubled checked everything, eslint also didn't found it. Any ideas what it might be?

**Pull request merged**